src/scripting/flash/text/textrenderer.cpp

```diff
@@ -0,0 +1,72 @@
+/**************************************************************************
+    Lightspark, a free flash player implementation
+
+    Copyright (C) 2009-2013  Alessandro Pignotti (a.pignotti@sssup.it)
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+**************************************************************************/
+
+#include "scripting/flash/text/textrenderer.h"
+#include "scripting/class.h"
+#include "scripting/argconv.h"
+#include <iomanip>
+
+using namespace lightspark;
+
+void TextRenderer::sinit(Class_base* c)
+{
+	CLASS_SETUP_NO_CONSTRUCTOR(c, ASObject, CLASS_SEALED | CLASS_FINAL);
+
+	c->setDeclaredMethodByQName("setAdvancedAntiAliasingTable","",Class<IFunction>::getFunction(c->getSystemState(),_setAdvancedAntiAliasingTable),NORMAL_METHOD,false);
+
+    c->setDeclaredMethodByQName("displayMode","",Class<IFunction>::getFunction(c->getSystemState(),_getDisplayMode),GETTER_METHOD,false);
+	c->setDeclaredMethodByQName("displayMode","",Class<IFunction>::getFunction(c->getSystemState(),_setDisplayMode),SETTER_METHOD,false);
+
+    c->setDeclaredMethodByQName("maxLevel","",Class<IFunction>::getFunction(c->getSystemState(),_getMaxLevel),GETTER_METHOD,false);
+	c->setDeclaredMethodByQName("maxLevel","",Class<IFunction>::getFunction(c->getSystemState(),_setMaxLevel),SETTER_METHOD,false);
+}
+
+ASFUNCTIONBODY_ATOM(TextRenderer,_setAdvancedAntiAliasingTable)
+{
+	LOG(LOG_NOT_IMPLEMENTED,"TextRenderer.setAdvancedAntiAliasingTable is a stub");
+	tiny_string fontName, colorType;
+	_NR<Array> advancedAntiAliasingTable;
+	ARG_UNPACK_ATOM(fontName)(colorType)(advancedAntiAliasingTable);
+}
+
+ASFUNCTIONBODY_ATOM(TextRenderer,_getDisplayMode)
+{
+    TextRenderer* th = asAtomHandler::as<TextRenderer>(obj);
+    ret = asAtomHandler::fromString(sys,th->displayMode);
```

Comment:
This will crash because getDisplayMode is static, that means obj is never an instance of TextRenderer.
It's better to just use local variables here and add a NOT_IMPLEMENTED message for now.
Same for the other getters/setters in this class.

Reply:
Cool, I've pushed further changes up but have not tested changes out. I'll improve my test and I'll then resolve t_his issue in particular_ tomorrow :)